sql/fxa_derived/fxa_amplitude_export_v1/query.sql

```diff
@@ -0,0 +1,128 @@
+      ) AS user_id,
+      MIN(CONCAT(insertId, '-user')) AS insert_id,
+      CAST(@submission_date AS DATETIME) AS timestamp,
+    -- Amplitude properties, scalars
+      `moz-fx-data-shared-prod`.udf.mode_last(ARRAY_AGG(jsonPayload.fields.region)) AS region,
+      `moz-fx-data-shared-prod`.udf.mode_last(ARRAY_AGG(jsonPayload.fields.country)) AS country,
+      `moz-fx-data-shared-prod`.udf.mode_last(
+        ARRAY_AGG(jsonPayload.fields.`language`)
+      ) AS `language`,
+    -- Event properties, arrays
+      ARRAY_AGG(
+        DISTINCT JSON_EXTRACT_SCALAR(jsonPayload.fields.event_properties, "$.service") IGNORE NULLS
+      ) AS services,
+      ARRAY_AGG(
+        DISTINCT JSON_EXTRACT_SCALAR(
+          jsonPayload.fields.event_properties,
+          "$.oauth_client_id"
+        ) IGNORE NULLS
+      ) AS oauth_client_ids,
+    -- User properties, arrays
+      ARRAY_AGG(
+        DISTINCT JSON_EXTRACT_SCALAR(
+          jsonPayload.fields.user_properties,
+          "$['$append'].fxa_services_used"
+        ) IGNORE NULLS
+      ) AS fxa_services_used,
+      ARRAY_AGG(DISTINCT jsonPayload.fields.os_name IGNORE NULLS) AS os_used_day,
+      CAST([] AS ARRAY<STRUCT<arr ARRAY<STRING>>>) AS os_used_week,
+      CAST([] AS ARRAY<STRUCT<arr ARRAY<STRING>>>) AS os_used_month,
+    -- User properties, scalars
+      MAX(
+        CAST(JSON_EXTRACT_SCALAR(jsonPayload.fields.user_properties, "$.sync_device_count") AS INT64)
+      ) AS sync_device_count,
+      MAX(
+        CAST(JSON_EXTRACT_SCALAR(jsonPayload.fields.user_properties, "$.sync_active_devices_day") AS INT64)
+      ) AS sync_active_devices_day,
+      MAX(
+        CAST(JSON_EXTRACT_SCALAR(jsonPayload.fields.user_properties, "$.sync_active_devices_week") AS INT64)
+      ) AS sync_active_devices_week,
+      MAX(
+        CAST(JSON_EXTRACT_SCALAR(jsonPayload.fields.user_properties, "$.sync_active_devices_month") AS INT64)
+      ) AS sync_active_devices_month,
+      `moz-fx-data-shared-prod`.udf.mode_last(
+        ARRAY_AGG(
+          JSON_EXTRACT_SCALAR(jsonPayload.fields.user_properties, "$.ua_version") IGNORE NULLS
+        )
+      ) AS ua_version,
+      `moz-fx-data-shared-prod`.udf.mode_last(
+        ARRAY_AGG(
+          JSON_EXTRACT_SCALAR(jsonPayload.fields.user_properties, "$.ua_version") IGNORE NULLS
+        )
+      ) AS ua_browser,
+      MAX(CAST(jsonPayload.fields.app_version AS FLOAT64)) AS app_version,
+      CAST(TRUE AS INT64) AS days_seen_bits,
+    FROM
+      base_events
+    GROUP BY
+      user_id
+  ),
+  _previous AS (
+    SELECT
+      * EXCEPT (submission_timestamp)
+    FROM
+      fxa_derived.fxa_amplitude_export_v1
+    WHERE
+      DATE(submission_timestamp) = DATE_SUB(@submission_date, INTERVAL 1 DAY)
+      AND udf.shift_28_bits_one_day(days_seen_bits) > 0
+  )
+
+SELECT
+  CAST(@submission_date AS TIMESTAMP) AS submission_timestamp,
+  _current.* REPLACE (
```

Comment:
Note that if there are no values for that user for that day, then all of the fields of `_current` will be empty. The view will filter those rows out, but we can continue to join with that data for future days.